diff --git a/includes/VEParsoid/src/Rest/Handler/ParsoidHandler.php b/includes/VEParsoid/src/Rest/Handler/ParsoidHandler.php
index 649a9aa..2a30582 100644
--- a/includes/VEParsoid/src/Rest/Handler/ParsoidHandler.php
+++ b/includes/VEParsoid/src/Rest/Handler/ParsoidHandler.php
@@ -361,7 +361,7 @@ abstract class ParsoidHandler extends Handler {
 
 		// FIXME there should be a better way to do this
 		global $wgRestPath;
-		$path = wfExpandUrl( "$wgRestPath$path", PROTO_CURRENT );
+		$path = wfExpandUrl( "$wgRestPath$path", PROTO_HTTP );
 
 		// FIXME this should not be necessary in the REST entry point
 		unset( $queryParams['title'] );
